src/promnesia/sources/viber_desktop.py

```diff
@@ -0,0 +1,164 @@
+"""
+Adapted from `telegram.py` to read from `~/.ViberPC/XYZ123/viber.db`
+"""
+
+import json
+import logging
+import textwrap
+from pathlib import Path
+from typing import Optional, TypeVar, Union
+from urllib.parse import \
+    unquote  # TODO mm, make it easier to rememember to use...
+
+from ..common import (Loc, PathIsh, Results, Visit, echain, extract_urls,
+                      from_epoch, get_logger)
+
+# TODO potentially, belongs to my. package
+
+# TODO kython?
+T = TypeVar("T")
+
+logger = logging.getLogger(__name__)
+
+
+def unwrap(res: Union[T, Exception]) -> T:
+    if isinstance(res, Exception):
+        raise res
+    else:
+        return res
+
+
+# TODO move to common?
+def dataset_readonly(db: Path):
+    # see https://github.com/pudo/dataset/issues/136#issuecomment-128693122
+    import sqlite3
+
+    import dataset  # type: ignore
+
+    creator = lambda: sqlite3.connect(f"file:{db}?immutable=1", uri=True)
+    return dataset.connect("sqlite:///", engine_kwargs={"creator": creator})
+
+
+def index(database: PathIsh) -> Results:
```

Comment:
One nice thing to do could be to make it optional , and it it's None, then do glob over `"~/.ViberPC/*/viber.db"`, then the user could just specify `Source(viber_desktop.index)` in the config without extra arguments.
But up to you.

Reply:
Nice idea!

Reply:
Implemented in 2d5c3c4.